flutter/lib/src/span_frame_metrics_collector.dart

```diff
@@ -0,0 +1,212 @@
+import 'dart:collection';
+
+import 'package:flutter/cupertino.dart';
+import 'package:meta/meta.dart';
+
+import '../sentry_flutter.dart';
+
+import 'frame_callback_handler.dart';
+import 'native/sentry_native.dart';
+
+@internal
+class SpanFrameMetricsCollector implements PerformanceContinuousCollector {
+  final SentryFlutterOptions options;
+  final FrameCallbackHandler? _frameCallbackHandler;
+  final SentryNative? _native;
+
+  /// Stores frame timestamps and their durations in milliseconds.
+  /// Keys are frame timestamps, values are frame durations.
+  final frameDurations = SplayTreeMap<DateTime, int>();
+
+  /// Stores the spans that are actively being tracked.
+  /// After the frames are calculated and stored in the span the span is removed from this list.
+  final activeSpans = <ISentrySpan>[];
+
+  bool get isTrackingPaused => _isTrackingPaused;
+  bool _isTrackingPaused = true;
+
+  bool get isTrackingRegistered => _isTrackingRegistered;
+  bool _isTrackingRegistered = false;
+
+  final _stopwatch = Stopwatch();
+
+  SpanFrameMetricsCollector(this.options,
+      {FrameCallbackHandler? frameCallbackHandler, SentryNative? native})
+      : _frameCallbackHandler =
+            frameCallbackHandler ?? DefaultFrameCallbackHandler(),
+        _native = native ?? SentryFlutter.native;
+
+  @override
+  void onSpanStarted(ISentrySpan span) {
+    if (span is NoOpSentrySpan || !options.enableFramesTracking) {
+      return;
+    }
+
+    activeSpans.add(span);
+    startFrameTracking();
+  }
+
+  @override
+  Future<void> onSpanFinished(ISentrySpan span, DateTime endTimestamp) async {
+    if (span is NoOpSentrySpan || !options.enableFramesTracking) {
```

Comment:
if `enableFramesTracking` is false the span was never added, so no need to check here, as the user could change the option at runtime (even if it's unsupported)
```suggestion
    if (span is NoOpSentrySpan) {
```

Reply:
the flag isn't used anywhere else except here, or maybe should we guard adding the performance collector behind it in SentryFlutter.init?

Reply:
i meant that the user could do `options.enableFramesTracking = true` anytime at runtime, even if this shouldn't be done. So checking this flag when the span is added is enough, as if it wasn't added, this function will do nothing.
You can even change it to `span is NoOpSentrySpan || !runningSpans.contains(span)`

Reply:
oh yeah makes sense, sry misunderstood, I thought the change referred to the `onSpanStarted` function